calculator/main.py

```diff
@@ -58,18 +89,39 @@ def get_history(calc):
         ie:
         ('2016-05-20 12:00:00', 'add', (1, 2), 3),
     """
-    pass
+    return calc['history']
 
 
 def reset_history(calc):
     """
     Resets the calculator history back to an empty list.
     """
-    pass
+    calc['history'] = []
+    return calc
 
 
 def repeat_last_operation(calc):
     """
     Returns the result of the last operation executed in the history.
     """
-    pass
+    #'perform_operation(calc, operation, params)'
```

Comment:
Since the last member in each history entry in `calc['history']` is the result of the operation, and since all `repeat_last_operation()` returns is the result, it is not necessary to run `perform_operations()` again here.  You could just return `calc['history'][-1][3]`.